src/java.base/share/classes/jdk/internal/util/ByteArray.java

```diff
@@ -41,12 +40,21 @@ public final class ByteArray {
     private ByteArray() {
     }
 
-    private static final VarHandle SHORT = create(short[].class);
-    private static final VarHandle CHAR = create(char[].class);
-    private static final VarHandle INT = create(int[].class);
-    private static final VarHandle FLOAT = create(float[].class);
-    private static final VarHandle LONG = create(long[].class);
-    private static final VarHandle DOUBLE = create(double[].class);
+    /**
+     * The {@code Unsafe} can be functionality replaced by
+     * {@linkplain java.lang.invoke.MethodHandles#byteArrayViewVarHandle byteArrayViewVarHandle},
+     * but it's not feasible in practices, because {@code ByteArray} and {@code ByteArrayLittleEndian}
+     * can be used in fundamental classes, {@code VarHandle} exercise many other
+     * code at VM startup, this could lead a recursive calls when fundamental
+     * classes is used in {@code VarHandle}.
```

Comment:
This comment is confusing. If this code is changed then the comment can be very simple to say that it uses Unsafe to allow it be used in early startup and the VarHandle implementation.